Plugin.pm

```diff
@@ -1208,8 +1244,30 @@ sub _trackItem {
 		$item->{line1} .= ' (' . cstring($client, 'PLUGIN_QOBUZ_HIRES') . ')';
 	}
 
-	if ( $track->{work} ) {
-		$item->{work} = $track->{work};
+	# Enhancements to work/composer display for classical music (tags returned from Qobuz are all over the place)
+	if ( $track->{album}->{isClassique} ) {
+		if ( $track->{work} ) {
+			$item->{work} = $track->{work};
+		} else {
+			# Try to extract the work from the title
+			if ( $track->{composer}->{name} && $track->{title} ) {
+				my @titleSplit = split /:\s*/, $track->{title};	
+				if ( index($track->{composer}->{name}, $titleSplit[0]) == -1 ) {
+					$item->{work} = $titleSplit[0];
+				} elsif ( $titleSplit[1] ) {
+					$item->{work} = $titleSplit[1];
+				}
+			}
+			$item->{line1} =~ s/$item->{work}://;
+		}
+		$item->{displayWork} = $item->{work};
+		$item->{displayWork} = $track->{composer}->{name} . string('COLON') . ' ' . $item->{work} if ($track->{composer}->{name});
+		$item->{work} = $track->{composer}->{name} . string('COLON') . ' ' . $item->{work} if ($track->{composer}->{name});
+		if ( $track->{composer}->{name} ) {
+			my $composerSurname = (split ' ', $track->{composer}->{name})[-1];
+			$item->{line1} =~ s/$composerSurname://;
```

Comment:
use `\Q` before using a variable in a regex

Reply:
Done

Reply:
But not pushed yet?

Reply:
Pushed now. Twice! ;)